x/distribution/keeper/msg_server.go

```diff
@@ -243,6 +243,11 @@ func (k msgServer) WithdrawTokenizeShareRecordReward(goCtx context.Context, msg
 	if err != nil {
 		return nil, err
 	}
+
+	if k.bankKeeper.BlockedAddr(ownerAddr) {
```

Comment:
Please consider moving this check to the `WithdrawTokenizeShareRecordReward` function. It contains all the validation for the procedure. It should be near the top of the function, right after fetch/unmarshal steps.